qutip/solver/solver_base.py

```diff
@@ -394,3 +395,98 @@ def add_integrator(cls, integrator, key):
                             " of `qutip.solver.Integrator`")
 
         cls._avail_integrators[key] = integrator
+
+
+def _solver_deprecation(kwargs, options, solver="me"):
+    """
+    Function to help the transition from v4 to v5.
+    Raise warnings for solver input that where moved from parameter to options.
+    """
+    if options is None:
+        options = {}
+    # TODO remove by 5.1
+    if "progress_bar" in kwargs:
+        warnings.warn(
+            '"progress_bar" is now included in options:\n'
+            'Use `options={"progress_bar": False / True / "tqdm" / "enhanced"}`'
+        )
+        options["progress_bar"] = kwargs.pop("progress_bar")
+
+    if "_safe_mode" in kwargs:
+        warnings.warn(
+            '"_safe_mode" is no longer supported.'
+        )
+        del kwargs["_safe_mode"]
+
+    if "verbose" in kwargs and solver == "br":
+        warnings.warn(
+            '"verbose" is no longer supported.'
+        )
+        del kwargs["verbose"]
+
+    if "tol" in kwargs and solver == "br":
+        warnings.warn(
+            'The "tol" parameter is no longer used. '
+            '`qutip.settings.core["auto_tidyup_atol"]` '
+            'is now used for rounding small values in sparse arrays.'
+        )
+        del kwargs["tol"]
+
+    if "map_func" in kwargs and solver in ["mc", "stoc"]:
+        warnings.warn(
+            '"map_func" is now included in options:\n'
+            'Use `options={"map": "serial" / "parallel" / "loky"}`'
+        )
+        # 4.X pass the function, 5.X use a string
+        del kwargs["map_func"]
```

Comment:
We could try the function's `__name__` here and use it if it looks right. Perhaps not worth the complexity though.

Reply:
I made it so function are also accepted in v5.